aten/src/ATen/native/cuda/Math.cuh

```diff
@@ -108,14 +108,14 @@ static inline __host__ __device__ scalar_t calc_digamma(scalar_t in) {
 
   accscalar_t x = static_cast<accscalar_t>(in);
   if (x == 0) {
-    return static_cast<scalar_t>(INFINITY);
+    return std::copysign(static_cast<scalar_t>(INFINITY), -x);
```

Comment:
Same comments for CUDA, too

Reply:
Is the static_cast needed for INFINITY here? It doesn't appear in the cpu code.

Reply:
For CPU, there are two functions for float and double

For CUDA, there is one which is templatized function.

Reply:
No worries. Let's just leave it.